tests/platforms/kubernetes/appmanager.go

```diff
@@ -196,6 +202,53 @@ func (m *AppManager) ValidiateSideCar() (bool, error) {
 	return true, nil
 }
 
+// DoPortForwarding performs port forwarding to access test apps in the cluster
+func (m *AppManager) DoPortForwarding() error {
+	podClient := m.client.Pods(m.namespace)
+	// Filter only 'testapp=appName' labeled Pods
+	podList, err := podClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", TestAppLabelKey, m.app.AppName),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	var podName string
+
+	for _, pod := range podList.Items {
+		podName = pod.Name
+		break
+	}
+
+	config:= m.client.GetClientConfig()
+
+	// create spdy roundtripper
+	roundTripper, upgrader, err := spdy.RoundTripperFor(config)
+	if err != nil {
+		return err
+	}
+
+	// create stop and ready channels
+	stopChan, readyChan := make(chan struct{}, 1), make(chan struct{}, 1)
+	out, errOut := new(bytes.Buffer), new(bytes.Buffer)
+
+	path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", m.namespace, podName)
+	hostIP := strings.TrimLeft(config.Host, "https:/")
+	serverURL := url.URL{Scheme: "https", Path: path, Host: hostIP}
+
+	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: roundTripper}, http.MethodPost, &serverURL)
+
+	// TODO keeping local port as 8080, not sure, so waiting for review
+	ports := []string{fmt.Sprintf("%d:%d", "8080", m.app.AppPort)}
```

Comment:
Can it be an env variable?

Reply:
@shalabhms We need to have a logic to find the free port on localhost because a test can use multiple portforward.

BTW, doesn't it require the methods to stop port-forwarding? `Dispose()` method should stop all port-forwards(so we may need to maintain the list of port-forwarded ports) for this particular apps during cleaning up.

Reply:
@shalabhms you can use [this](https://github.com/phayes/freeport) package to get a free port or a list of free ports. It's used by the Dapr CLI.

Reply:
> Can it be an env variable?

thanks @artursouza  for review. I am now using @yaron2  suggested freeport package.

Reply:
> @shalabhms We need to have a logic to find the free port on localhost because a test can use multiple portforward.
> 
> BTW, doesn't it require the methods to stop port-forwarding? `Dispose()` method should stop all port-forwards(so we may need to maintain the list of port-forwarded ports) for this particular apps during cleaning up.

Now added the stop forwarding in the dispose.

Reply:
> @shalabhms you can use [this](https://github.com/phayes/freeport) package to get a free port or a list of free ports. It's used by the Dapr CLI.

Thanks. I added this freeport pacakge now. thanks !